readme.md

```diff
@@ -4,14 +4,75 @@ Code repository for the Goose VI general platform board for the electrical archi
 
 ## Installation
 
-```
+```bash
 git clone https://github.com/waterloop/G6-MCU-Platform-Board.git
 cd G6-MCU-Platform-Board
 ```
 
-## Build
+## Building
+
+Each supported device has its own corresponding directory. To build for a particular device,
+pass the device directory into `make` using the `DEV` variable.
+
+For example:
+
+```bash
+make DEV=STM32G473CBTx
+```
+
+The build will generate the following:
+1. Object files contained in `build/objects`
+2. A standalone executable in `build/standalone`
+3. A static library in `build/platform` to be linked with an application
+
+## Developing for This Repository
+
+### Intellisense and Autocomplete
+Intellisense and autocomplete are available through the `clangd` language server. It is supported on
+most modern editors.
+
+If you use VSCode, there exists an
+[extension](https://marketplace.visualstudio.com/items?itemName=llvm-vs-code-extensions.vscode-clangd)
+for it.
+
+`clangd` is already installed on your [vagrant VMs](https://github.com/waterloop/firmware-vagrant).
+It also easily installed through apt and brew if you like.
+
+`clangd` uses the auto-generated file `compile_commands.json` to provide intellisense and autocomplete.
+This file is gitignored in this repository, so you will need to generate it yourself if you want
+intellisense. This can be easily done using `bear`, which is pre-installed on your vagrant VMS.
 
 ```
-cd /path/to/G6-MCU-Platform-Board
-make
+# this is an example, can replace DEV with whatever device you need
+bear -- make DEV=STM32G473CBTx
 ```
+
+### Code
+
+Write your code in the `platform` directory.
+
```

Comment:
So the goal for this is to be common drivers and the "application code" will be for board specific code? For example, BMS code would be application code? but can driver would be platform code?

Reply:
Not exactly. I guess, all the code that is common to the MCU board would be in the platform layer (e.g., CAN, EEPROM, etc).

The makefile in this repo will build a static library `build/platform/platform.a` which can be linked with the application code.